server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java

```diff
@@ -57,21 +58,40 @@ public QProfileFactory(DbClient db) {
   // ------------- CREATION
 
   QualityProfileDto getOrCreate(DbSession dbSession, OrganizationDto organization, QProfileName name) {
-    Objects.requireNonNull(organization, "Organization is required, when creating a quality profile.");
+    requireNonNull(organization);
     QualityProfileDto profile = db.qualityProfileDao().selectByNameAndLanguage(organization, name.getName(), name.getLanguage(), dbSession);
     if (profile == null) {
       profile = doCreate(dbSession, organization, name);
     }
     return profile;
   }
 
-  public QualityProfileDto create(DbSession dbSession, OrganizationDto organization, QProfileName name) {
-    Objects.requireNonNull(organization, "Organization is required, when creating a quality profile.");
+  /**
+   * Create the quality profile in DB with the specified name.
+   *
+   * @throws BadRequestException if a quality profile with the specified name already exists
+   */
+  public QualityProfileDto checkAndCreate(DbSession dbSession, OrganizationDto organization, QProfileName name) {
+    requireNonNull(organization);
     QualityProfileDto dto = db.qualityProfileDao().selectByNameAndLanguage(organization, name.getName(), name.getLanguage(), dbSession);
     checkRequest(dto == null, "Quality profile already exists: %s", name);
     return doCreate(dbSession, organization, name);
   }
 
+  /**
+   * Create the quality profile in DB with the specified name.
+   *
+   * A DB error will be thrown if the quality profile already exists.
+   */
+  public QualityProfileDto create(DbSession dbSession, OrganizationDto organization, QProfileName name) {
+    return doCreate(dbSession, requireNonNull(organization), name);
+  }
+
+  private static OrganizationDto requireNonNull(@Nullable OrganizationDto organization) {
+    Objects.requireNonNull(organization, "Organization is required, when creating a quality profile.");
+    return organization;
```

Comment:
could be inlined but can't be statically imported because this method has the same name and signature as one of the override of `Objects.requireNonNull`